firstvoices/backend/models/page.py

```diff
@@ -0,0 +1,65 @@
+from django.core.validators import validate_slug
+from django.db import models
+from django.db.models import Q
+from django.utils.translation import gettext as _
+
+from backend.models import Image
+from backend.models.base import BaseControlledSiteContentModel
+from backend.models.media import Video
+from backend.models.widget import SiteWidgetList
+from backend.permissions import predicates
+
+
+class SitePage(BaseControlledSiteContentModel):
+    class Meta:
+        verbose_name = _("site page")
+        verbose_name_plural = _("site pages")
+        rules_permissions = {
+            "view": predicates.is_visible_object,
+            "add": predicates.is_superadmin,  # permissions will change when we add a write API
+            "change": predicates.is_superadmin,
+            "delete": predicates.is_superadmin,
+        }
+        constraints = [
+            # Constraint to ensure a SitePage has either a banner image or video or none but not both.
+            models.CheckConstraint(
+                name="SitePage_banner_image_or_video_only_one",
+                check=(
+                    Q(
+                        Q(banner_image=None, banner_video=None)
+                        | Q(Q(banner_image=None) & ~Q(banner_video=None))
+                        | Q(~Q(banner_image=None) & Q(banner_video=None))
+                    )
+                ),
+            )
+        ]
+
+    title = models.CharField(max_length=225)
+    slug = models.SlugField(
+        max_length=200,
+        blank=False,
+        validators=[validate_slug],
+        db_index=True,
+        unique=True,
+    )
```

Comment:
@gmcauliffe Do we currently restrict the page urls at all, to skip reserved ones like "dictionary" etc?

Reply:
There is a custom page route "sandbox" of sorts in the frontend to prevent conflicts with any of our FirstVoices paths like "dictionary".  A list of reserved paths in the BE would be great (if it isn't too time consuming) and would help to prevent any potential confusion caused by duplicates like this:
"smalgyax/dictionary" and "smalgyax/custom-page/dictionary.

On a separate point, We definitely want paths to be unique within a site but NOT unique across all sites (i.e. two different language sites should be able to have the same path e.g. "smalgyax/language-tools" and "SECOTEN/language-tools").

Thanks for raising this :)

Reply:
Thanks for the feedback!

I have added a [`unique_together`](https://github.com/First-Peoples-Cultural-Council/fv-be/blob/b3d0a7b33bba2293b8dd4394d59bdddef0e4c3ba/firstvoices/backend/models/page.py#L31) Meta option to the model which will ensure that the site page slugs are unique across a site but can be the same on two different sites. 

I have also [created a validator](https://github.com/First-Peoples-Cultural-Council/fv-be/blob/b3d0a7b33bba2293b8dd4394d59bdddef0e4c3ba/firstvoices/backend/models/validators.py#L1) that checks that site page slugs do not contain any of the strings in the reserved strings list. @gmcauliffe would you be able to point me to the current frontend list of reserved paths so I can match them up? If there are any others that we might want to add I can do that as well.

Another option here would be to store reserved slug strings in a model that can be updated without pushing code changes. If others have thoughts about this I am open to suggestions. @sarahfirstvoices

Reply:
I think listing the reserved slugs here is great for mvp, and we can consider making it configurable in the future.